test/Microsoft.AspNetCore.Mvc.FunctionalTests/AntiforgeryTestHelper.cs

```diff
@@ -2,37 +2,44 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.IO;
 using System.Linq;
 using System.Net.Http;
-using System.Xml.Linq;
+using AngleSharp.Dom.Html;
+using AngleSharp.Parser.Html;
 
 namespace Microsoft.AspNetCore.Mvc.FunctionalTests
 {
     public static class AntiforgeryTestHelper
     {
+        public static string RetrieveAntiforgeryToken(string htmlContent)
+            => RetrieveAntiforgeryToken(htmlContent, actionUrl: string.Empty);
+
         public static string RetrieveAntiforgeryToken(string htmlContent, string actionUrl)
         {
-            htmlContent = "<Root>" + htmlContent + "</Root>";
-            var reader = new StringReader(htmlContent);
-            var htmlDocument = XDocument.Load(reader);
+            var parser = new HtmlParser();
+            var htmlDocument = parser.Parse(htmlContent);
```

Comment:
This parser should handle implied CDATA elements e.g. `<script>` and DTD declarations better than `XDocument`. But, I'm curious if anything elsewhere in this PR broke the previous approach?

Reply:
I backed out a couple of tests that passed the `IHtmlDocument` in, but in general seems like the correct thing to use a Html parser to parse html.

Reply:
Agreed. Just confirming.